src/proxy.rs

```diff
@@ -192,8 +198,8 @@ mod tests {
 
     #[cfg(feature = "socks-proxy")]
     #[test]
-    fn parse_proxy_socks_user_pass_server_port() {
-        let proxy = Proxy::new("socks://user:p@ssw0rd@localhost:9999").unwrap();
+    fn parse_proxy_socks4_user_pass_server_port() {
+        let proxy = Proxy::new("socks4://user:p@ssw0rd@localhost:9999").unwrap();
```

Comment:
I think we need to duplicate this test to have a `socks://` ensure SOCKS5 and a `socks4://` doing SOCKS4.

To maintain backwards compatibility and not have to major version, we need to make it so `socks://` means SOCKS5. This should also be highlighted in the docs.

Reply:
> To maintain backwards compatibility and not have to major version, we need to make it so socks:// means SOCKS5.

`socks` still means SOCKS5

Reply:
Cool! thanks!